### PR TITLE
Move proper tail calls runtime out of runtime.js

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

**Comment by arv** [comment]
Only last commit

**Comment by arv** [comment]
@johnjbarton ping

File: src/runtime/proper-tail-calls.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    setupProperTailCalls();
+  }
+  setPrivate(func, isTailRecursiveName, true);
+  return func;
+}
+
+$traceurRuntime.initTailRecursiveFunction = initTailRecursiveFunction;
+$traceurRuntime.call = tailCall;
+$traceurRuntime.continuation = createContinuation;
+$traceurRuntime.construct = construct;
```
Comment (johnjbarton) [inline src/runtime/proper-tail-calls.js]: How about an IIFE around this file?

**Comment by johnjbarton** [comment]
LGTM

File: src/runtime/proper-tail-calls.js (hunk 1 above)
Comment (arv) [inline src/runtime/proper-tail-calls.js]: This is a `Module`.

Comment (johnjbarton) [inline src/runtime/proper-tail-calls.js]: too much bootstrapping for me today ;-)

**Pull request closed without merging**